test/test_foreach.py

```diff
@@ -537,6 +537,18 @@ def test_add_scalar_with_empty_list_and_empty_tensor(self, device, dtype):
             # Regression test for https://github.com/pytorch/pytorch/issues/113156
             torch._foreach_mul_(tensors, 1)
 
+    @dtypes(*all_types_and_complex_and(torch.half, torch.bfloat16))
+    def test_add_with_size_one_value_and_stride_not_equal(self, device, dtype):
+        tensor = torch.ones((2, 1, 3, 3), device=device, dtype=dtype)
+        tensor_stride = torch.ones((2, 1, 3, 3), device=device, dtype=dtype).as_strided(
+            (2, 1, 3, 3), (9, 1, 3, 1)
+        )
+        left_input = [tensor, tensor_stride]
+        right_input = [tensor_stride, tensor]
+        compare_result = tensor + tensor_stride
+        for res in torch._foreach_add(left_input, right_input):
+            self.assertEqual(res, compare_result)
```

Comment:
Hmmm...this test cases passes for me locally without the change. I haven't figured out why but either this test case is not testing what we want or it is already checked.

Reply:
ok, without ignore dims of one, add op will be called for each input. I will using ForeachFuncWrapper to check this.